engine/src/main/java/org/terasology/utilities/subscribables/SpecificSubscribable.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.utilities.subscribables;
+
+import java.beans.PropertyChangeListener;
+
+/**
+ * Originally created to provide Observable-like functionality to config classes,
+ * it can be used by any class that wishes to be monitored for change in specific properties.
+ */
+public interface SpecificSubscribable {
```

Comment:
I would try to keep the number of interfaces and abstract base classes as small as needed. How about renaming `SpecificSubscribable` to `Subscribable` interface and `AbstractSpecificSubscribable` to `SubscribableConfig` (the `transient` makes it somewhat unique) and removing the other ones?